libraries/standard/http/src/http_client.c

```diff
@@ -682,40 +758,40 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
 
 /*-----------------------------------------------------------*/
 
-static HTTPStatus_t sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
-                                     const HTTPRequestHeaders_t * pRequestHeaders )
+static HTTPStatus_t sendHttpData( const HTTPTransportInterface_t * pTransport,
+                                  const uint8_t * pBuffer,
+                                  size_t bufferLen )
 {
     HTTPStatus_t returnStatus = HTTP_SUCCESS;
     int32_t transportStatus = 0;
 
     assert( pTransport != NULL );
     assert( pTransport->send != NULL );
-    assert( pRequestHeaders != NULL );
+    assert( pBuffer != NULL );
 
-    /* Send the HTTP headers over the network. */
     transportStatus = pTransport->send( pTransport->pContext,
-                                        pRequestHeaders->pBuffer,
-                                        pRequestHeaders->headersLen );
+                                        pBuffer,
+                                        bufferLen );
 
     if( transportStatus < 0 )
     {
-        LogError( ( "Failed to send HTTP headers: Transport send()"
+        LogError( ( "Failed to send HTTP data: Transport send()"
```

Comment:
With this common utility for all `send` functions, the use of the generic `data` label in the logs reduces the effectiveness of the log. One way to know the specific segment of the HTTP request being sent is by passing a string parameter to the function from each of its callers as `"headers"`, "body"`, `"content-length"`

Reply:
I log in debug before entering this function what we are sending.